src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4338,6 +4341,15 @@ BoundExpression bindObjectCreationExpression(ObjectCreationExpressionSyntax node
             }
         }
 
+        private BoundExpression BindCollectionCreationExpression(CollectionCreationExpressionSyntax syntax, BindingDiagnosticBag diagnostics)
+        {
+            if (MessageID.IDS_FeatureCollectionLiterals.GetFeatureAvailabilityDiagnosticInfo((CSharpParseOptions)syntax.SyntaxTree.Options) is { } diagnosticInfo)
```

Comment:
can't this be: `MessageID.IDS_FeatureCollectionLiterals.CheckFeatureAvailability(diagnostics, syntax, syntax.OpenBracketToken.GetLocation())`?

Reply:
Yes, it can be. I was trying to avoid allocating the `Location` instance eagerly, but I can simplify.

Reply:
yeah, looksl ike we don't bother with that optimziation anywhere else.  if we did really want to do that though, we coudl have an overload that takes a SyntaxNodeOrToken and just realizes the location on demand if needed.

Reply:
Changed to use existing `CheckFeatureAvailability()` overload and added a PROTOTYPE comment to add `CheckFeatureAvailability(this MessageID feature, BindingDiagnosticBag diagnostics, SyntaxToken syntax)`.